src/main/java/pl/codzisnaobiad/imhungry/infrastructure/spoonacular/SpoonacularClient.java

```diff
@@ -52,11 +60,28 @@ SpoonacularSearchRecipesResponse searchRecipes(RecipeRequestModel recipeRequestM
                 .build()
                 .toUri();
 
-        HttpEntity<String> response = http.getForEntity(uri, String.class);
-        setQuotaPoints(response.getHeaders());
+        var response = executeGetRequest(uri);
         return mapJsonToObject(response.getBody(), SpoonacularSearchRecipesResponse.class);
     }
 
+    SpoonacularGetRecipeInformationResponse getRecipeInformationById(int id) {
+        var uri = fromHttpUrl(baseUrl)
+            .pathSegment(RECIPES_SEGMENT, String.valueOf(id), INFORMATION_SEGMENT)
+            .queryParam(INCLUDE_NUTRITION_PARAM, INCLUDE_NUTRITION)
+            .queryParam(API_KEY_PARAM, apiKey)
+            .build()
+            .toUri();
+
+        var response = executeGetRequest(uri);
```

Comment:
Tu można by obsłużyć typowe wyjątki przy zapytaniu http i zrobić wreszcie exception handlera xd

Reply:
to osobno xd